url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/Demo/PreparedDemo.java

```diff
@@ -0,0 +1,237 @@
+          handleInsert((MInsert)st);
+        } else if(st instanceof MUpdate) {
+          handleUpdate((MUpdate)st);
+        } else if(st instanceof MDelete) {
+          handleDelete((MDelete)st);
+        }
+      }
+
+    } catch(Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   */
+  static void handleQuery(MQuery q) throws Exception {
+    System.out.println("SELECT Statement:");
+
+    Vector sel = q.getSelect(); // SELECT part of the query
+
+    MExpression w = (MExpression)q.getWhere();
+    if(w != null) {
+
+      Hashtable meta = null;
+
+      Vector from = q.getFrom();  // FROM part of the query
+
+      // Column metadata: Hashtable (alias, table name)
+      // Will be used by handleWhere() to resolve aliases into table names
+      meta = new Hashtable();
+      for(int i=0; i<from.size(); i++) {
+        MFromItem fi = (MFromItem)from.elementAt(i);
+        String alias = fi.getAlias();
+        if(alias == null) alias = fi.getTable();
+        meta.put(alias.toUpperCase(), fi.getTable());
+      }
+
+      handleWhere(w, meta);
+    }
+
+  }
+
+  static void handleInsert(MInsert ins) throws Exception {
+    System.out.println("INSERT Statement:");
+    String tab = ins.getTable();
+    Vector values = ins.getValues();
+    if(values == null) {
+      System.out.println("no VALUES(), probably a subquery ?");
+    }
+    int nval = values.size();
+    Vector columns = ins.getColumns();
+    if(columns == null) {
+      System.out.println("no column names, assuming _col_1"
+       + (nval > 1 ? " to _col_" + nval : ""));
+      columns = new Vector(nval);
+      for(int i=1; i<=nval; i++) {
+        columns.addElement("_col_" + i);
+      }
+    }
+
+    for (int i=0; i<nval; i++) {
+      MExp v = (MExp)values.elementAt(i);
+      if(isPreparedColumn(v)) {
+        System.out.println("[" + tab + "," + columns.elementAt(i) + "]");
+      }
+    }
+  }
+
+  static void handleUpdate(MUpdate upd) throws Exception {
+    System.out.println("UPDATE Statement:");
+
+    String tab = upd.getTable();
+    Hashtable set = upd.getSet();
+    Enumeration k = set.keys();
+    while(k.hasMoreElements()) {
+      String col = (String)k.nextElement();
+      MExp e = (MExp)set.get(col);
+      if(isPreparedColumn(e)) {
+        System.out.println("[" + tab + "," + col + "]");
+      }
+    }
+
+    MExpression w = (MExpression)upd.getWhere();
+    if(w != null) {
+      Hashtable meta = new Hashtable(1);
+      meta.put(tab, tab);
+      handleWhere(w, meta);
+    }
+
+  }
+
+  static void handleDelete(MDelete del) throws Exception {
+    System.out.println("DELETE Statement:");
+
+    String tab = del.getTable();
+
+    MExpression w = (MExpression)del.getWhere();
+    if(w != null) {
+      Hashtable meta = new Hashtable(1);
+      meta.put(tab, tab);
+      handleWhere(w, meta);
+    }
+
+  }
+
+  static void handleWhere(MExp e, Hashtable meta) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.